operator/controllers/execution/scans/hook_reconciler.go

```diff
@@ -62,250 +60,232 @@ func (r *ScanReconciler) setHookStatus(scan *executionv1.Scan) error {
 	return nil
 }
 
-func (r *ScanReconciler) executeReadAndWriteHooks(scan *executionv1.Scan) error {
-	// Get the first Hook Status which is not completed.
-	ctx := context.Background()
-	var nonCompletedHook *executionv1.HookStatus
-
-	for _, hook := range scan.Status.ReadAndWriteHookStatus {
+func getNonCompletedHookPriorityQueue(hooks *[]*executionv1.HookStatus) util.PriorityQueue {
+	var priorityQueue = make(util.PriorityQueue, len(*hooks))
+	var completedHooks = 0
+	for _, hook := range *hooks {
 		if hook.State != executionv1.Completed {
-			nonCompletedHook = &hook
-			break
+			priorityQueueItem := util.PriorityQueueItem{
+				Value:    hook,
+				Priority: hook.Priority,
+			}
+			priorityQueue[completedHooks] = &priorityQueueItem
+			completedHooks++
 		}
 	}
+	priorityQueue = priorityQueue[:completedHooks]
+	heap.Init(&priorityQueue)
```

Comment:
The heap data structure should be an implementation detail of the priority queue (encapsulation).